src/databases/functions/users/user-unblock-by-admin.js

```diff
@@ -0,0 +1,12 @@
+const {User} = require('../../models');
+
+module.exports = async (userId) => {
+  if (!userId) return null;
+  try {
+    const result = await User.update({blocked_by_admin: false}, {where: {user_id: userId}});
+    return result;
+  } catch (error) {
+    console.error(`Failed to unblock user with id ${userId}: `, error);
+    return null;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `userUnblockByAdmin` is well written and handles the case where `userId` might be null. However, it's not a good practice to return `null` when an error occurs. Instead, you should throw the error so that the caller can handle it appropriately. Also, logging errors directly in the function may not be ideal for testing purposes. Consider using an error handling middleware or a logger service.

```diff
-    console.error(`Failed to unblock user with id ${userId}: `, error);
-    return null;
+    throw new Error(`Failed to unblock user with id ${userId}: ${error}`);
```
This way, the error can be caught and handled by the calling function, allowing for more flexible error handling.


<!-- This is an auto-generated comment by OSS CodeRabbit -->